examples/cluster/plot_segmentation_toy.py

```diff
@@ -51,8 +51,9 @@
 circle3 = (x - center3[0]) ** 2 + (y - center3[1]) ** 2 < radius3**2
```

Comment:
I suggest adding a heading on line 33. Something like:
```diff
-import numpy as np

+# %%
+# Generate the data
+# -----------------
+
+import numpy as np
```

I also think it might be nice for us to move the imports from this cell to the top of the cell in which they are first used.